compiler/AST/WhileDoStmt.cpp

```diff
@@ -39,7 +46,59 @@ BlockStmt* WhileDoStmt::build(Expr* cond, BlockStmt* body)
   {
     retval = CForLoop::buildCForLoop(toCallExpr(cond), body);
   }
+  else if (CallExpr* condIV = isPrimIfVar(cond))
+  {
+    // while var X = condExpr do ...
+    DefExpr* varDef = toDefExpr(condIV->get(1)->remove());
+    Expr*  condExpr = condIV->get(1)->remove();
+    INT_ASSERT(! varDef->init && ! varDef->exprType); // from parser
+    
+    // The construction follows the non-IfVar case below, adding the following:
+    //
+    //  // at start of 'retval' block
+    //  def while_borrow = chpl_checkBorrowIfVar(condExpr, true)
+    //  ... use 'while_borrow' where the non-IfVar cases uses 'cond' ...
+    //
+    //  // at start of 'loop' body
+    //  def X
+    //  move X <- PRIM_TO_NON_NILABLE_CLASS(while_borrow)
+    //
+    //  // at end of 'loop' body
+    //  move while_borrow, chpl_checkBorrowIfVar(condExpr, true)
+    //  ... use 'while_borrow' where the non-IfVar cases uses 'cond' ...
+
+    // todo: share code with the non-IfVar case (below)
```

Comment:
Would you mind leaving your name and the date on this TODO in case we forget to get back to it quickly?  Obviously I'm hoping it will happen soon, but life happens :)